File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    #b = format(value, 'b')
    b1 = format(n, 'b')
    l = b1.count('1')
    answer = n+1    
    while True:
        if(l == format(answer, 'b').count('1')):
            break
        answer += 1        
        
    return answer

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int currC = cc(Integer.toBinaryString(n));
        
        while(true){
            n++;
            if(currC == cc(Integer.toBinaryString(n))) break;
        }
        
        return n;
    }
    
    public int cc(String s){
        int c = 0;
        for(char cc : s.toCharArray()){
            if(cc=='1') c++;
        }
        return c;
    }
}